.circleci/config.yml

```diff
@@ -650,42 +665,66 @@ jobs:
             - run:
                 name: Ensure package.json Version And Tag Match
                 command: cd desktop/bin && node validate_tag.js $VERSION
+      - run: *desktop-decrypt-assets
       - run:
-          name: Install Node
+          name: Compute cache checksums
           command: |
-            set +e
-            set +x
-            export NVM_DIR="/opt/circleci/.nvm"
-            [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
-            nvm install
-            nvm use
+            # Get latest commit that modified packages directory
+            echo "$(git log -n 1 --format='%h' -- packages .nvmrc yarn.lock)" > packages-hash
+
+            # Get latest commit that modified .css and related files
+            echo "$(git log -n 1 --format='%h' -- '*.css' '*.scss' '*.sass')" > webpack-css-loader-hash
+      - restore_cache: *desktop-restore-yarn-cache
+      - restore_cache: *desktop-restore-packages-cache
+      - restore_cache: *desktop-restore-webpack-css-loader-cache
+      - run:
+          name: Install Linux deps
+          command: |
+            sudo apt update
+            sudo apt-get install -y git-restore-mtime bc
+      - run:
+          name: Restore file modified time
+          command: |
+            # Webpack cache-loader depends on file mtime, which is reset by "git clone".
+            # Use git-restore-mtime to restore original modification time of files based
+            # on date of most recent commit that modified them.
+            # Ref: http://manpages.ubuntu.com/manpages/bionic/man1/git-restore-mtime.1.html
+            #
+            # Note: this utility is highly optimized and is able to process all Calypso
+            # files in less than 10 seconds!
+            /usr/lib/git-core/git-restore-mtime --force --commit-time --skip-missing
       - run:
           name: Install Dependencies
           command: |
-            export NVM_DIR="/opt/circleci/.nvm"
+            # Install Node
+            # Note: this syntax is required to load NVM_DIR in the machine executor.
+            # Ref: https://discuss.circleci.com/t/circleci-forgetting-node-version-on-machine-executor/28813
+            set +e
+            set +x
             [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+            nvm install
             nvm use
             npm install -g yarn
 
-            yarn config set chromedriver_skip_download true
-            yarn config set chromedriver_filepath '/usr/local/bin/chromedriver'
-
-            yarn install --frozen-lockfile
-      # - run: *yarn-install
-      - restore_cache: *desktop-restore-source-cache
-      - run: *desktop-decrypt-assets
+            # defer to restored packages whenever possible
+            if [ -f packages-cache-restored ]; then
+              export DISABLEPOSTINSTALL=1
+            fi
+            yarn install --frozen-lockfile --cache-folder ~/.cache/yarn
```

Comment:
Nitpick: You can as well specify the yarn cache folder as an environment variable (`YARN_CACHE_FOLDER`)

Reply:
Nice - didn't know about this, thanks!